android/expoview/src/main/java/host/exp/exponent/RNObject.kt

```diff
@@ -0,0 +1,314 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package host.exp.exponent
+
+import host.exp.exponent.analytics.EXL
+import host.exp.expoview.BuildConfig
+import java.lang.reflect.Constructor
+import java.lang.reflect.Field
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+
+// TODO: add type checking in DEBUG
+class RNObject {
+  private val className: String // Unversioned
+  private var clazz: Class<*>? = null // Versioned
+  private var instance: Any? = null // Versioned
+
+  // We ignore the version of clazz
+  constructor(clazz: Class<*>?) {
+    className = removeVersionFromClass(clazz)
+  }
+
+  constructor(className: String) {
+    this.className = className
+  }
+
+  private constructor(obj: Any?) {
+    assign(obj)
+    className = removeVersionFromClass(clazz)
+  }
+
+  val isNull: Boolean
+    get() = instance == null
+  val isNotNull: Boolean
+    get() = instance != null
+
+  // required for "unversioned" flavor check
+  fun loadVersion(version: String): RNObject {
+    try {
+      clazz = if (version == UNVERSIONED || BuildConfig.FLAVOR == "unversioned") {
+        if (className.startsWith("host.exp.exponent")) {
+          Class.forName("versioned.$className")
+        } else {
+          Class.forName(className)
+        }
+      } else {
+        Class.forName("abi" + version.replace('.', '_') + '.' + className)
+      }
+    } catch (e: ClassNotFoundException) {
+      EXL.e(TAG, e)
+    }
+    return this
+  }
+
+  fun assign(obj: Any?) {
+    if (obj != null) {
+      clazz = obj.javaClass
+    }
+    instance = obj
+  }
+
+  fun get(): Any? {
+    return instance
+  }
+
+  fun rnClass(): Class<*>? {
+    return clazz
+  }
+
+  fun version(): String {
+    val versionedClassName = clazz!!.name
+    return if (versionedClassName.startsWith("abi")) {
+      val abiVersion = versionedClassName.split(".").toTypedArray()[0]
+      abiVersion.substring(3)
+    } else {
+      UNVERSIONED
+    }
```

Comment:
Seems like this can just be:
```suggestion
    return RNObject.versionForClassname(clazz!!.name)
```